src/Ocelot.Provider.Kubernetes/Kube.cs

```diff
@@ -26,28 +30,39 @@ public class Kube : IServiceDiscoveryProvider
         _logger = factory.CreateLogger<Kube>();
         _kubeApi = kubeApi;
         _serviceBuilder = serviceBuilder;
-        _services = new();
     }
 
     public virtual async Task<List<Service>> GetAsync()
     {
-        var endpoint = await _kubeApi
-            .ResourceClient(client => new EndPointClientV1(client))
-            .GetAsync(_configuration.KeyOfServiceInK8s, _configuration.KubeNamespace);
+        var endpoint = await Retry.OperationAsync(GetEndpoint, Ensure, logger: _logger);
 
-        _services.Clear();
-        if (endpoint?.Subsets.Count != 0)
+        if (Ensure(endpoint))
         {
-            _services.AddRange(BuildServices(_configuration, endpoint));
+            _logger.LogWarning(() => GetMessage($"Unable to use bad result returned by {nameof(Kube)} integration endpoint because the final result is invalid/unknown after multiple retries!"));
+            return new(0);
```

Comment:
You could use collection expression ```[]``` here instead of ```new(0)```

Reply:
Due to the lack of support for [Collection expressions](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions) in .NET 6/7, we will encounter warnings in our build logs. I've experienced this issue before with testing projects, where a large number of such warnings caused my confusion. [Collection expressions](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions) were introduced in [C# 12](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12) for .NET 8. Opening the solution in Visual Studio 2018 or earlier versions will likely result in a build failure. Only Visual Studio 2022 version 17.8 or later can recognize **Collection Expressions** and compile them without warnings or errors. However, even with VS v17.8+, targeting `net7.0` and `net6.0` frameworks still generates warnings in the logs. It appears we must wait for the [end-of-support](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core) of .NET 6/7 to fully utilize this feature of C# 12. My apologies! We will only utilize .NET 6 features until November 2024. 😉 Following the release of .NET 9, we will begin using the new feature as we migrate our solution to .NET 8/9. Is that okay?